src/FluentIterator.php

```diff
@@ -0,0 +1,331 @@
+
+
+use phpDocumentor\Reflection\Types\Static_;
+use Traversable;
+
+class FluentIterator implements \IteratorAggregate
+{
+    /**
+     * @var iterable
+     */
+    private $iterator;
+
+    public function __construct(iterable $iterator)
+    {
+        $this->iterator = $iterator;
+    }
+
+    public function getIterator()
+    {
+        return $this->iterator;
+    }
+
+    /**
+     * @see \iter\map()
+     */
+    public function map(callable $function): self
+    {
+        return new static(map($function, $this->iterator));
+    }
+
+    /**
+     * @see \iter\mapKeys()
+     */
+    public function mapKeys(callable $function): self
+    {
+        return new static(mapKeys($function, $this->iterator));
+    }
+
+    /**
+     * @see \iter\flatMap()
+     */
+    public function flatMap(callable $function): self
+    {
+        return new static(flatMap($function, $this->iterator));
+    }
+
+    /**
+     * @see \iter\reindex()
+     */
+    public function reindex(callable $function): self
+    {
+        return new static(reindex($function, $this->iterator));
+    }
+
+    /**
+     * @see \iter\apply()
+     */
+    public function apply(callable $function): void
+    {
+        apply($function, $this->iterator);
+    }
+
+    /**
+     * @see \iter\apply()
+     */
+    public function filter(callable $function): self
+    {
+        return new static(filter($function, $this->iterator));
+    }
+
+    /**
+     * @see \iter\enumerate()
+     */
+    public function enumerate(): self
+    {
+        return $this->toPairs();
+    }
+
+    /**
+     * @see \iter\toPairs()
+     */
+    public function toPairs(): self
+    {
+        return new static(toPairs($this->iterator));
+    }
+
+    /**
+     * @see \iter\fromPairs()
+     */
+    public function fromPairs(): self
+    {
+        return new static(fromPairs($this->iterator));
+    }
+
+    /**
+     * @see \iter\reduce()
+     */
+    public function reduce(callable $function, $startValue = null)
+    {
+        return reduce($function, $this->iterator, $startValue);
+    }
+
+    /**
+     * @see \iter\reductions()
+     */
+    public function reductions(callable $function, $startValue = null): self
+    {
+        return new static(reductions($function, $this->iterator, $startValue));
+    }
+
+    /**
+     * Combines this iterable for keys and another for values into one iterator.
+     *
+     * Examples:
+     *
+     *     new FluentIterator(['a', 'b', 'c'])->zipKeys([1, 2, 3])
+     *     => iter('a' => 1, 'b' => 2, 'c' => 3)
+     *
+     * @param iterable $values Iterable of values
+     *
+     * @return \Iterator
+     */
+    public function zipValues(iterable $values): self
+    {
+        return new static(zipKeyValue($this->iterator, $values));
+    }
+
+    /**
+     * Combines this iterable for values and another for keys into one iterator.
+     *
+     * Examples:
+     *
+     *     new FluentIterator(['a', 'b', 'c'])->zipKeys([1, 2, 3])
```

Comment:
```suggestion
     *     (new FluentIterator(['a', 'b', 'c']))->zipKeys([1, 2, 3])
```

...though a ctor function would be better.